File: src/main/java/CodingPractise/Array/OddEvenSeparateSegregate.py
#http://www.geeksforgeeks.org/segregate-even-and-odd-numbers/
class OddEvenSeparate:
    def __init(self):
        pass
    def separate(self,mylist):
        print "in",mylist
        start=0;end=len(mylist)-1
        while start<end:
            print start
            while(mylist[start]%2==0 and start<end):
                start=start+1
            while(mylist[end]%2==1 and start<end):
                end=end-1;
            print "start",start
            print "end",end
            if (start<end):
                temp=mylist[start]
                mylist[start]=mylist[end]
                mylist[end]=temp
                start=start+1
                end=end-1
        return mylist


    def separate_smallest_code(self,mylist):
        for ele in range(len(mylist)):
            if mylist[ele] % 2 == 0:
                even=mylist.pop(ele)
                mylist.insert(0,even)
        return mylist



if __name__=='__main__':
    # mylist=[-99,21,20,11,2000,51,61,-100]
    mylist=[99,21,20,11,2000,51,61,100,91,92, 93,95,97,101]
    sp=OddEvenSeparate()
    print sp.separate(mylist)
    print sp.separate_smallest_code(mylist)

File: src/main/java/CodingPractise/Array/OddEvenSeparateSegregate.java
package CodingPractise.Array;

import java.util.Arrays;
//http://www.geeksforgeeks.org/segregate-even-and-odd-numbers/

/**
 * Created by muthuselvan on 9/28/17.
 */
public class OddEvenSeparateSegregate {
    public static void main(String[] args) {
        System.out.println(Arrays.toString(segregateEvenOdd(new int[]{1, 34, 45, 9, 8, 90, 3})));
//        System.out.println(Arrays.toString(segregateEvenOdd(new int[]{2, 1, 4,9})));
//        System.out.println(Arrays.toString(segregateEvenOdd(new int[]{5,2,9,12})));


    }

    static int[] segregateEvenOdd(int arr[])
    {
        /* Initialize left and right indexes */
        int left = 0, right = arr.length - 1;
        while (left < right)
        {
            /* Increment left index while we see 0 at left */
            while (arr[left]%2 == 0 && left < right)
                left++;
            System.out.println("No of even nos : " +left);

            /* Decrement right index while we see 1 at right */
            while (arr[right]%2 == 1 && left < right)
                right--;

            System.out.println("No of odd nos : " +right);

            if (left < right)
            {
                /* Swap arr[left] and arr[right]*/
                int temp = arr[left];
                arr[left] = arr[right];
                arr[right] = temp;
                left++;
                right--;
            }
        } // End outer while loop
        return arr;
    }



}
